### DISCUSSION TITLE
Can `BUNDLE_FROZEN` override a `.bundle/config`: `BUNDLE_FROZEN: "true"`

### DISCUSSION THREAD

**Discussion by stdedos** [ruby/rubygems#7510] [Q&A] [answered]
And, in general, how does the final configuration compiled with?

**Comment by deivid-rodriguez** [comment]
See the configuration precedence in Bundler documentation here: https://bundler.io/v2.5/man/bundle-config.1.html.

Cheers!

**Reply by stdedos** [reply]
> Local config (<project_root>/.bundle/config or $BUNDLE_APP_CONFIG/config)
> Environmental variables (ENV)

So ... theoretically it does, right?
"I think" I wanted to report a situation that it doesn't work. In particular with Renovatebot, I think

We have a "workaround" like

```ruby
Bundler.settings.temporary(frozen: false)
```

but on CI that _that_ script is not invoked, _this_ workaround does not get applied - hence it does not work.

-----

But this is since March, therefore ... I don't remember much 🤷

**Reply by deivid-rodriguez** [reply, marked as answer]
Yeah, sorry for the delay. In principle, a local `.bundle/config`file should take precedence over `ENV`, but a global `~/.bundle/config` should not.

If this is working differently, it's unexpected, yes.

**Reply by stdedos** [reply]
> In principle, a local `.bundle/config`file should take precedence over `ENV`

I expected/said the opposite. So, I guess "all is working correctly".
